### DISCUSSION TITLE
Json example doesn't work if the loop repeats?

### DISCUSSION THREAD

**Discussion by d-floyd** [xlate/staedi#270] [Q&A] [answered]
I ran your test code for reading an edi stream into json: https://github.com/xlate/staedi/wiki/Reading-EDI-Data-(JSON-Conversion). I got the same response w/ the same payload/schema. However,  when I repeated the loop structure in the edi payload -- I only got the last entry of the repeating elements.  Looking for verification that my understanding/expectation is correct... and the example code has an issue (as opposed to the code is correct and my understanding/input it flawed).

Expected result.. json payload as in the example except with "2000" as an array  like this.
`            "2000": [
             {
                "AK2": ["837","0021"],
                "2100": {
                    "AK3": ["NM1","8","","8"],
                    "AK4": [ ["8"],"66", "7", "MI" ]
                },
                "AK5": ["R","5"]
            },			
             {
                "AK2": ["838","0021"],
                "2100": {
                    "AK3": ["NM1","9","","8"],
                    "AK4": [ ["8"],"66", "7", "MI" ]
                },
                "AK5": ["R","5"]
            }			
            ],`

edi payloaded I tested with:

`ISA*00*          *00*          *ZZ*ReceiverID     *ZZ*Sender         *050812*1953*^*00501*508121953*0*P*:~
GS*FA*ReceiverDept*SenderDept*20050812*195335*000005*X*005010X230~
ST*997*0001~
AK1*HC*000001~
AK2*837*0021~
AK3*NM1*8**8~
AK4*8*66*7*MI~
AK5*R*5~
AK2*838*0021~
AK3*NM1*9**8~
AK4*8*66*7*MI~
AK5*R*5~
AK9*R*1*1*0~
SE*8*0001~
GE*1*000005~
IEA*1*508121953~`

**Comment by MikeEdgar** [comment, marked as answer]
@d-floyd you're right, that example doesn't work with repeated loops because it uses the loop ID as the key. If you wanted to make that work it would probably be better to represent loops as arrays rather than objects.

**Comment by NandaBalakrishnan** [comment]
@MikeEdgar , i am facing  same issue with repeated loops, can you share some sample code to make it work?

**Reply by MikeEdgar** [reply]
@NandaBalakrishnan you will want to make the start/end loop events look more like the start/end transaction events. That will give you a loop entry that itself is an array of objects. I intend to update the example page but I have not yet had time.

**Reply by NandaBalakrishnan** [reply]
@d-floyd, we are also facing this issue, do you have any reference code to fix above issue?

**Reply by MikeEdgar** [reply]
@NandaBalakrishnan - similar to how the example creates a `JsonArrayBuilder` for `transactions`, you'll need to do the same for loops. Rather than having a single builder, you will need to create one for each loop and push/pop them on and off a stack as the start/end loop events occur.
